### ISSUE TITLE
Logic of op training order

### ISSUE THREAD

**Issue by dragen1860** [brain-research/acai#9]
Dear author:
  I found a potential logic problem:
  In baseline.py, Line 58: https://github.com/brain-research/acai/blob/6f26d685b0fc4cf89afcdebb58d370c9a3374fdb/baseline.py#L58
, you train `train_op = train_op.minimize(loss + xloss,
                                         tf.train.get_global_step())` meanwhile. But from my personal perspective, I think you should train `loss` op for some iterations and THEN train a single layer classifier wit `xloss` op to test the classification accuracy.
Do you feel the same way? Thanks.

**Comment by david-berthelot** [comment]
In our paper, I believe we mentioned that we trained the classifier in parallel. As long as we are consistent in doing so across the various auto-encoders, and we are, I don't see why this would be problem.
Can you further detail your argument?

**Comment by craffel** [comment]
Indeed, we wrote

> To test different autoencoders in this setting, we trained a separate single-layer classifier *in tandem* with the autoencoder using the latent representation as input.

As David says, this should be the same for all autoencoders, and we only
compared our own implementation (and didn't claim state of the art compared
to anything), so I don't think this is an important distinction.

**Comment by dragen1860** [comment]
OK, it's good you made it clear to me.
Indeed i have a technical problem to seek for assistance from you.
I found it's difficult to implement this scenario: an training phase existed in a train_op chain.
For your case, let's say, from operation a => b => loss => train_op, I want to embed an extra training phase in `a=>b`, that's: `a=>training for k steps=>b=>loss=>train_op`. I dnt know how to implement this.
It seems having a nested training stage.

**Comment by david-berthelot** [comment]
I think that for general programming/tensorflow questions, may be stackexchange or the tensorflow github are better places than here. Closing this issue as our code is behaving as expected.
